worlds/yoshisisland/__init__.py

```diff
@@ -266,25 +260,24 @@ def get_excluded_items(self) -> Set[str]:
 
     def create_item_with_correct_settings(self, player: int, name: str) -> Item:
         data = item_table[name]
-        classification = self.item_classifications[data.classification]
-        item = Item(name, classification, data.code, player)
+        item = Item(name, data.classification, data.code, player)
 
         if not item.advancement:
             return item
 
-        if name == 'Car Morph' and self.options.stage_logic.value != 0:
+        if name == 'Car Morph' and not self.options.stage_logic:
             item.classification = ItemClassification.useful
 
-        if name == 'Secret Lens' and (self.options.hidden_object_visibility.value >= 2 or self.options.stage_logic.value != 0):
+        if name == 'Secret Lens' and (self.options.hidden_object_visibility >= 2 or self.options.stage_logic != 0):
             item.classification = ItemClassification.useful
 
-        if name in ["Bonus 1", "Bonus 2", "Bonus 3", "Bonus 4", "Bonus 5", "Bonus 6", "Bonus Panels"] and self.options.minigame_checks.value <= 1:
+        if name in ["Bonus 1", "Bonus 2", "Bonus 3", "Bonus 4", "Bonus 5", "Bonus 6", "Bonus Panels"] and self.options.minigame_checks <= 1:
             item.classification = ItemClassification.useful
 
-        if name in ["Bonus 1", "Bonus 3", "Bonus 4", 'Bonus Panels'] and self.options.item_logic.value == 1:
+        if name in ["Bonus 1", "Bonus 3", "Bonus 4", 'Bonus Panels'] and self.options.item_logic == 1:
             item.classification = ItemClassification.progression
 
-        if name == 'Piece of Luigi' and self.options.goal.value != 0:
+        if name == 'Piece of Luigi' and self.options.goal != 0:
```

Comment:
Don't need the != 0 here

Reply:
For understanding what the code is doing,
is it not more relevant to check that the goal is Luigi hunt, rather that checking that it isn't Bowser?

It seems like that would also make the code more future-proof, in case another goal is added to this option in the future.

I think this would be my preference,
- `.value` is a matter of preference. I like "explicit is better than implicit"
- `== PlayerGoal.option_luigi_hunt` for someone reading the code to understand what it's checking.
```suggestion
        if name == 'Piece of Luigi' and self.options.goal.value == PlayerGoal.option_luigi_hunt:
```